### PR TITLE
[frontendproxy] enable envoy environment resource detector

### PR DESCRIPTION
**Pull request by rlankfo**
# Changes

This enables the environment resource detector for the envoy opentelemetry tracer. See https://github.com/envoyproxy/envoy/pull/29547

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [x] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: src/frontendproxy/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,7 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
-
-FROM envoyproxy/envoy:v1.28-latest
+FROM envoyproxy/envoy:dev-698abe98131fa5c64904082407878d1fcc62cb88
```
Comment (julianocosta89) [inline src/frontendproxy/Dockerfile:4]: Would you mind waiting for the stable release, unless using the `dev` one?

**Comment by puckpuck** [comment]
I noticed envoy still hasn't created a release that includes the required functionality. No harm in keeping this open and around as a reminder for when they finally release.

File: src/frontendproxy/Dockerfile (hunk 1 above)
Comment (julianocosta89) [reply src/frontendproxy/Dockerfile:4]: @rlankfo I saw that `v1.29.0` was released, would you mind updating the PR and adding an entry in the changelog for this?

Comment (rlankfo) [reply src/frontendproxy/Dockerfile:4]: absolutely! thanks for the heads up.

**Comment by puckpuck** [comment]
Thanks for this @rlankfo 

Can you explain how one would use this resource detector? Is it via the OTEL_RESOURCE_ATTRIBUTES env var on the envoy container?

**Comment by rlankfo** [comment]
> Is it via the OTEL_RESOURCE_ATTRIBUTES env var on the envoy container?

Yes, this is correct. Here's the official documentation: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/tracers/opentelemetry/resource_detectors/v3/environment_resource_detector.proto#extension-envoy-tracers-opentelemetry-resource-detectors-environment

**Review by cedricziel — APPROVED** [review]
@rlankfo 👋 

ℹ️ I added the environment var to the service definitions in docker compose. 

LGTM!

Before:

Bildschirmfoto 2024-01-19 um 09 19 56

After:

Bildschirmfoto 2024-01-19 um 09 25 56

**Comment by julianocosta89** [comment]
The Helm chart actually needs to be updated.
The env var also needs to be added there:
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-demo/values.yaml#L358-L386

**Review by julianocosta89 — APPROVED** [review]
LGTM

**Review by puckpuck — APPROVED** [review]

**Pull request merged**